node/sc/sub_bridge_handler.go

```diff
@@ -342,6 +383,46 @@ func (sbh *SubBridgeHandler) LocalChainHeadEvent(block *types.Block) {
 	}
 }
 
+func (sbh *SubBridgeHandler) handleParentChainInvalidTxResponseMsg(msg p2p.Msg) error {
+	/*
+		///////////////////// KIP71 Test ///////////////////////////
+		sbh.subbridge.blockchain.Config().KIP71CompatibleBlock = big.NewInt(100)
+		headerNum := big.NewInt(101)
+		////////////////////////////////////////////////////////////
+	*/
+
+	headerNum := sbh.subbridge.blockchain.CurrentHeader().Number
+	var invalidTxs []InvalidParentChainTx
+	if err := msg.Decode(&invalidTxs); err != nil && err != rlp.EOL {
+		return errResp(ErrDecode, "msg %v: %v", msg, err)
+	}
+	txPool := sbh.subbridge.GetBridgeTxPool()
+	for _, invalidTx := range invalidTxs {
+		if tx := txPool.Get(invalidTx.TxHash); tx != nil {
+			logger.Error("A bridge tx was not executed", "err", invalidTx.ErrStr,
+				"prevGasPrice", sbh.subbridge.bridgeAccounts.GetParentGasPrice(), "txHash", invalidTx.TxHash.String(),
+				"txGasPrice", tx.GasPrice().Uint64(),
+				"txHash", tx.Hash().String())
+			if sbh.subbridge.blockchain.Config().IsKIP71ForkEnabled(headerNum) &&
+				invalidTx.ErrStr == blockchain.ErrGasPriceBelowBaseFee.Error() {
+				logger.Info("Request gasPrice and KIP71 values to parent chain")
+				sbh.SyncNonceAndGasPrice()
+
+				logger.Error("Bridge tx is removed which has lower gasPrice than UpperBoundBaseFee")
+				// Remove the tx and delegate re-execution of the tx by Value Transfer Recovery feature
+				if err := sbh.subbridge.GetBridgeTxPool().RemoveTx(tx); err != nil {
```

Comment:
Is it possible to perform vtrecovery logic with just removing? I'm just curious.

Reply:
1. The value transfer recovery (VTR) finds unhandled events from the counterpart chain. As the event was not handled by the parent chain, the VTR finds unhandled events and regenerates the corresponding txs.
2. The received invalid txs must be removed because they live in bridgeTxPool forever and consumes unnecessary resending overhead. (i.e., `txPending` is counted)

Reply:
Can you tell me where should I look in to see the code related to #1 (what you explained above)?
I'm not familiar with service chain logic, so i want to know how to regenerate when tx is removed from bridgeTxPool.